app/src/main/java/nerd/tuxmobil/fahrplan/congress/alarms/AlarmReceiver.java

```diff
@@ -55,7 +58,7 @@ public void onReceive(Context context, Intent intent) {
             NotificationCompat.Builder builder = notificationHelper.getSessionAlarmNotificationBuilder(contentIntent, title, when, soundUri);
             boolean isInsistentAlarmsEnabled = appRepository.readInsistentAlarmsEnabled();
             MyApp.LogDebug(LOG_TAG, "Preference 'isInsistentAlarmsEnabled' = " + isInsistentAlarmsEnabled + ".");
-            notificationHelper.notify(NotificationHelper.SESSION_ALARM_ID, builder, isInsistentAlarmsEnabled);
+            notificationHelper.notify(new Random().nextInt(), builder, isInsistentAlarmsEnabled);
```

Comment:
This is giving up on engineering. You're resorting to a hack. If you want to leave it like this please at least add a comment explaining the approach.

I'd add a notification ID column to the database and use that to make sure your notification IDs are unique. This information could also be used to recreate notifications e.g. after a reboot.

Reply:
I stumbled about the fixed ID while working on some other tasks. I did not see any use case to keep the ID so I decided a random integer will do the job. What do you have in mind so it does not look like a hack?

Storing the notification ID to recreate a notification is a new feature to me which should be discussed separately.

Reply:
If you manage notification IDs (as opposed to using random numbers) you'll have to persist the notification ID values because in low memory situations your process might be killed, but the notifications remain visible to the user. 
If you only kept notification IDs in memory you wouldn't know about the existing notifications when your process is restarted. So the next notification created would most likely overwrite an existing notification that hasn't been tapped or swiped away by the user yet.

Being able to recreate notifications after e.g. a reboot is an additional feature that comes almost for free once you persist notification IDs.

Reply:
Revised in b0d4b62a01a9cced5e158e52029f538e469fc89c.

Reply:
Revised once again in e647e11c9cba21a1dd811f504064e7994045592a. I added that the notification data is deleted once the notification is consumed (speaking: user tapped it, app resumed). When a notification is swiped away the associated database row remains.

Reply:
_Draft:_ I will look into [how to detect if the user swiped away a notification](https://stackoverflow.com/questions/12665131/how-to-detect-if-a-notification-has-been-dismissed).

Reply:
Dismissed notifications are now removed from the database. See :arrow_down:

On a first test it seems a bit unreliable - I will test more as soon as I have time.

Reply:
I found the error - again the delete notification intent has to be unique. Successfully tested on 36C3, Pixel 2, Android 10. - Ready for review.

Reply:
@cketti How do you think about this implemention?